core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala

```diff
@@ -790,7 +809,36 @@ class TimeSeriesShard(val ref: DatasetRef,
     shardStats.purgedPartitions.increment(numDeleted)
   }
 
-  def createFlushTask(flushGroup: FlushGroup): Task[Response] = {
+  /**
+    * Creates zero or more flush tasks (up to the number of flush groups) based on examination
+    * of the record container's ingestion time. This should be called before ingesting the container.
+    */
```

Comment:
Hmmm.  The createFlushTasks creates tasks, which are not actually executed until mapAsync.  We should add a comment that the tasks are designed not to be executed right away, instead they are executed after ingest method in a mapAsync.